server/tests/src/test/java/org/infinispan/server/rollingupgrade/DefaultRollingUpgradeTest.java

```diff
@@ -10,4 +22,48 @@ public void testDefaultSetting() throws InterruptedException {
       RollingUpgradeConfigurationBuilder builder = new RollingUpgradeConfigurationBuilder("15.2.0.Final", "15.2.1.Final");
       RollingUpgradeHandler.performUpgrade(builder.build());
    }
+
+   @Test
+   public void testHttpClient() throws Exception {
+      String cacheName = "rolling-upgrade";
+      TestUser user = TestUser.ADMIN;
+      RollingUpgradeConfigurationBuilder builder = new RollingUpgradeConfigurationBuilder("15.2.0.Final", "15.2.1.Final")
+            .nodeCount(3);
+      RestClientConfigurationBuilder restBuilder = new RestClientConfigurationBuilder();
+      restBuilder.security().authentication().enable().username(user.getUser()).password(user.getPassword());
+      AtomicBoolean clientUtilized = new AtomicBoolean(false);
```

Comment:
Maybe instead of a single boolean we have a ByRef.Integer that counts interactions to verify against?

Reply:
Updated.